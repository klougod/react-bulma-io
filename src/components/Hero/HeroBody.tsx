import React from 'react'
import PropTypes from 'prop-types'

export const HeroBody = ({ className, children }: any) => (
  <div className={`hero-body ${className}`}>{children}</div>
)

HeroBody.defaultProps = {
  className: ''
}

HeroBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default HeroBody
