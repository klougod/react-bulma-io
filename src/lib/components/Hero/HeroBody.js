import React from 'react'
import PropTypes from 'prop-types'

export const HeroBody = ({ className, children }) => (
  <div className={`hero-body ${className}`}>
    { children }
  </div>
)

HeroBody.defaultProps = {
  className: ''
}

HeroBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroBody