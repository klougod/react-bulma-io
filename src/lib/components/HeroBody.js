import React from 'react'
import PropTypes from 'prop-types'

export const HeroBody = props => {

  const {
    className,
    children
  } = props

  return (
    <div className={`hero-body ${className}`}>
      { children }
    </div>
  )
}

HeroBody.defaultProps = {
  className: ''
}

HeroBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroBody