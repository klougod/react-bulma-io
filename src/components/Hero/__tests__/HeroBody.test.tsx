import React from 'react'
import Enzyme from 'enzyme'

import HeroBody from '../HeroBody'

describe('HeroBody component test', function () {
  it('should render without crash without props', function () {
    const heroBodyMock = Enzyme.shallow(<HeroBody />)
    expect(heroBodyMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const heroBodyMockProps = {
      className: 'heroBody CLass',
    }
    const heroBodyMock = Enzyme.shallow(<HeroBody {...heroBodyMockProps} />)
    expect(heroBodyMock).toMatchSnapshot()
  })
})
