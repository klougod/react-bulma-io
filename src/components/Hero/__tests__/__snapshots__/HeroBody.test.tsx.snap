// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBody component test should render without crash with props 1`] = `
<div
  className="hero-body heroBody CLass"
/>
`;

exports[`HeroBody component test should render without crash without props 1`] = `
<div
  className="hero-body "
/>
`;
